### PR TITLE
Fix video/video.c:1458:52: error: implicit conversion from enumeration type 'enum v4l2_buf_type' to different enumeration type 'imgsensor_stream_type_t' (aka 'enum imgsensor_stream_type_e') [-Werror,-Wenum-conversion]

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
```
ret = g_video_sensor_ops->validate_frame_setting(type, nr_fmt, sf, &si);
       ~~~~~~~~~~~~~~~~~~                         ^~~~
```

## Impact
video validation

## Testing
Pass CI

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@jerpelea could you review the change?

File: drivers/video/video.c [hunk 1]
```diff
@@ -1455,7 +1455,8 @@ static int validate_frame_setting(enum v4l2_buf_type type,
   convert_to_imgsensorfmt(&vfmt[VIDEO_FMT_SUB], &sf[IMGSENSOR_FMT_SUB]);
   convert_to_imgsensorinterval(interval, &si);
 
-  ret = g_video_sensor_ops->validate_frame_setting(type, nr_fmt, sf, &si);
+  ret = g_video_sensor_ops->validate_frame_setting(
+             type == V4L2_BUF_TYPE_STILL_CAPTURE, nr_fmt, sf, &si);
```
Comment (pkarashchenko) [inline drivers/video/video.c:1459]: Let's better use
```
((type == V4L2_BUF_TYPE_VIDEO_CAPTURE) ?
     IMGSENSOR_STREAM_TYPE_VIDEO : IMGSENSOR_STREAM_TYPE_STILL
```
same as in `start_capture`

Comment (xiaoxiang781216) [reply drivers/video/video.c:1459]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
https://github.com/apache/incubator-nuttx/pull/7391

**Pull request merged**